File: python/acronym/acronym.py
def abbreviate(words):
    words = words.replace("-", " ")
    words = words.replace("_", "")
    word_lst = words.split(" ")
    output = ""
    for item in word_lst:
        if item == "":
            continue
        output += item[0]
    return output.upper()


print(abbreviate("Something - I made up from thin air"), "SIMUFTA")

File: go/acronym/acronym.go
package acronym

import "strings"

const testVersion = 3

func Abbreviate(str string) string {
	// convert dashes to spaces
	str = strings.Replace(str, "-", " ", -1)
	strList := strings.Split(str, " ")
	output := ""
	for _, word := range strList {
		output += strings.ToUpper(string(word[0]))
	}
	return output
}